src/llfs/page_tracer.test.cpp

```diff
@@ -0,0 +1,88 @@
+//#=##=##=#==#=#==#===#+==#+==========+==+=+=+=+=+=++=+++=+++++=-++++=-+++++++++++
+//
+// Part of the LLFS Project, under Apache License v2.0.
+// See https://www.apache.org/licenses/LICENSE-2.0 for license information.
+// SPDX short identifier: Apache-2.0
+//
+//+++++++++++-+-+--+----- --- -- -  -  -   -
+
+#include <llfs/page_tracer.hpp>
+#include <llfs/trace_refs_recursive.hpp>
+
+#include <gmock/gmock.h>
+#include <gtest/gtest.h>
+
+#include <unordered_set>
+#include <vector>
+
+namespace {
+using namespace batt::int_types;
+
+class MockPageTracer : public llfs::PageTracer
+{
+ public:
+  MOCK_METHOD(batt::StatusOr<batt::BoxedSeq<llfs::PageId>>, trace_page_refs,
+              (llfs::PageId from_page_id), (override, noexcept));
+};
+
+//=#=#==#==#===============+=+=+=+=++=++++++++++++++-++-+--+-+----+---------------
+// Test Plan:
+//  1. Mock a PageTracer and test its integration with trace_refs_recursive by creating a DAG of
+//  PageIds.
+
+TEST(PageTracerTest, TraceRefsRecursiveTest)
+{
+  MockPageTracer mock_tracer;
+
+  // Create a DAG of PageIds, where nodes have PageIds in the range [1, num_nodes]. At first, every
+  // node (page) in the graph is a root (no incoming edges/refs), and we erase from the root set as
+  // we create edges.
+  //
+  std::unordered_set<llfs::PageId, llfs::PageId::Hash> roots;
+  u16 num_nodes = 100;
```

Comment:
nit: `usize` is probably better to use here.  Yes, we probably can get away with 16-bits, but it ever so slightly increases the cognitive load on the reader by keeping the thought of "will there be an integer overflow?" in the back of their mind.

Reply:
Fixed!